pkg/types/baremetal/platform.go

```diff
@@ -39,7 +40,7 @@ type Host struct {
 	HardwareProfile string           `json:"hardwareProfile"`
 	RootDeviceHints *RootDeviceHints `json:"rootDeviceHints,omitempty"`
 	BootMode        BootMode         `json:"bootMode,omitempty"`
-	NetworkConfig   string           `json:"networkConfig,omitempty"`
+	NetworkConfig   *apiextv1.JSON   `json:"networkConfig,omitempty"`
```

Comment:
I guess the plus side of this is that you **could** accept a string here as well. I don't know if we want to, but you could.